zerver/actions/custom_profile_fields.py

```diff
@@ -199,6 +202,34 @@ def do_update_user_custom_profile_data_if_changed(
             field_value.save(update_fields=["value", "rendered_value"])
         else:
             field_value.save(update_fields=["value"])
+
+        # If the field is a select field, we need to convert the value to the
+        # user-facing text representation.
+        if field_value.field.field_type == CustomProfileField.SELECT:
+            field_data = orjson.loads(field_value.field.field_data)
+            old_value_user_facing = field_data.get(old_value, {}).get("text") if old_value else None
+            new_value_user_facing = (
+                field_data.get(custom_profile_field["value"], {}).get("text")
+                or custom_profile_field["value"]
+            )
+        else:
+            old_value_user_facing = old_value
+            new_value_user_facing = custom_profile_field["value"]
+
+        # If the field is a mentor field, we need to convert the array
+        # of values to array of user-facing user_name.
```

Comment:
Mentor is not a field type -- it's just an example field with the "user account" field type.